src/MongoDB/Cursor.c

```diff
@@ -117,6 +125,16 @@ static void php_phongo_cursor_iterator_rewind(zend_object_iterator *iter TSRMLS_
 	php_phongo_cursor_t        *cursor = cursor_it->cursor;
 	const bson_t               *doc;
 
+	/* If the cursor was never advanced (e.g. command cursor), do so now */
+	if (!cursor->advanced) {
+		cursor->advanced = true;
+
+		if (!phongo_cursor_advance_and_check_for_error(cursor->cursor TSRMLS_CC)) {
+			/* Exception should already have been thrown */
```

Comment:
Yes, but shouldn't you destroy the cursor now?

Reply:
Once we create a `MongoDB\Driver\Cursor` with the `mongoc_cursor_t`, the only place we destroy the libmongoc cursor is from `php_phongo_cursor_free_object()`.